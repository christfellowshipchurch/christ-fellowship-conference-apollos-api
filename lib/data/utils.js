"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.parseRockKeyValuePairs = void 0;

/*
 Splits up a Rock Key Value paired string where | splits pairs and ^ splits key and value
 Returns null if keyValueStr is null */
const parseRockKeyValuePairs = (keyValueStr, keyOverride = null, valueOverride = null) => {
  const key = keyOverride || 'key';
  const value = valueOverride || 'value';
  return keyValueStr ? keyValueStr.split('|').map(n => {
    const splt = n.split('^');
    let rtn = {};
    rtn[key] = splt[0] || '';
    rtn[value] = splt[1] || '';
    return rtn;
  }) : [];
};

exports.parseRockKeyValuePairs = parseRockKeyValuePairs;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL3NyYy9kYXRhL3V0aWxzLmpzIl0sIm5hbWVzIjpbInBhcnNlUm9ja0tleVZhbHVlUGFpcnMiLCJrZXlWYWx1ZVN0ciIsImtleU92ZXJyaWRlIiwidmFsdWVPdmVycmlkZSIsImtleSIsInZhbHVlIiwic3BsaXQiLCJtYXAiLCJuIiwic3BsdCIsInJ0biJdLCJtYXBwaW5ncyI6Ijs7Ozs7OztBQUFBOzs7QUFHTyxNQUFNQSxzQkFBc0IsR0FBRyxDQUFDQyxXQUFELEVBQWNDLFdBQVcsR0FBRyxJQUE1QixFQUFrQ0MsYUFBYSxHQUFHLElBQWxELEtBQTJEO0FBQy9GLFFBQU1DLEdBQUcsR0FBR0YsV0FBVyxJQUFJLEtBQTNCO0FBQ0EsUUFBTUcsS0FBSyxHQUFHRixhQUFhLElBQUksT0FBL0I7QUFFQSxTQUFPRixXQUFXLEdBQ2RBLFdBQVcsQ0FBQ0ssS0FBWixDQUFrQixHQUFsQixFQUNDQyxHQURELENBQ01DLENBQUQsSUFBTztBQUNWLFVBQU1DLElBQUksR0FBR0QsQ0FBQyxDQUFDRixLQUFGLENBQVEsR0FBUixDQUFiO0FBQ0EsUUFBSUksR0FBRyxHQUFHLEVBQVY7QUFFQUEsSUFBQUEsR0FBRyxDQUFDTixHQUFELENBQUgsR0FBV0ssSUFBSSxDQUFDLENBQUQsQ0FBSixJQUFXLEVBQXRCO0FBQ0FDLElBQUFBLEdBQUcsQ0FBQ0wsS0FBRCxDQUFILEdBQWFJLElBQUksQ0FBQyxDQUFELENBQUosSUFBVyxFQUF4QjtBQUVBLFdBQU9DLEdBQVA7QUFDRCxHQVRELENBRGMsR0FXZCxFQVhKO0FBWUQsQ0FoQk0iLCJzb3VyY2VzQ29udGVudCI6WyIvKlxuIFNwbGl0cyB1cCBhIFJvY2sgS2V5IFZhbHVlIHBhaXJlZCBzdHJpbmcgd2hlcmUgfCBzcGxpdHMgcGFpcnMgYW5kIF4gc3BsaXRzIGtleSBhbmQgdmFsdWVcbiBSZXR1cm5zIG51bGwgaWYga2V5VmFsdWVTdHIgaXMgbnVsbCAqL1xuZXhwb3J0IGNvbnN0IHBhcnNlUm9ja0tleVZhbHVlUGFpcnMgPSAoa2V5VmFsdWVTdHIsIGtleU92ZXJyaWRlID0gbnVsbCwgdmFsdWVPdmVycmlkZSA9IG51bGwpID0+IHtcbiAgY29uc3Qga2V5ID0ga2V5T3ZlcnJpZGUgfHwgJ2tleSdcbiAgY29uc3QgdmFsdWUgPSB2YWx1ZU92ZXJyaWRlIHx8ICd2YWx1ZSdcblxuICByZXR1cm4ga2V5VmFsdWVTdHJcbiAgICA/IGtleVZhbHVlU3RyLnNwbGl0KCd8JylcbiAgICAgIC5tYXAoKG4pID0+IHtcbiAgICAgICAgY29uc3Qgc3BsdCA9IG4uc3BsaXQoJ14nKVxuICAgICAgICBsZXQgcnRuID0ge31cblxuICAgICAgICBydG5ba2V5XSA9IHNwbHRbMF0gfHwgJydcbiAgICAgICAgcnRuW3ZhbHVlXSA9IHNwbHRbMV0gfHwgJydcblxuICAgICAgICByZXR1cm4gcnRuXG4gICAgICB9KVxuICAgIDogW11cbn0iXX0=