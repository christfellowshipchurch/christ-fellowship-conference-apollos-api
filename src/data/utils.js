/*
 Splits up a Rock Key Value paired string where | splits pairs and ^ splits key and value
 Returns null if keyValueStr is null */
export const parseRockKeyValuePairs = (keyValueStr, keyOverride = null, valueOverride = null) => {
  const key = keyOverride || 'key'
  const value = valueOverride || 'value'

  return keyValueStr
    ? keyValueStr.split('|')
      .map((n) => {
        const splt = n.split('^')
        let rtn = {}

        rtn[key] = splt[0] || ''
        rtn[value] = splt[1] || ''

        return rtn
      })
    : []
}