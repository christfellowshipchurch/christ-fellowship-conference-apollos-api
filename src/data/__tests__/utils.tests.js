import { parseRockKeyValuePairs } from '../utils'

describe('ParseRockKeyValuePairs', () => {
    /* Test different use cases for parsing Key Value pairs that get passed from Rock */
    it('parses a key value pair string with no overrides', () => {
        const str = 'firstKey^firstValue|secondKey^secondValue'

        expect(parseRockKeyValuePairs(str)).toMatchSnapshot()
    })

    it('parses a key value pair string with key and value labeling overrides', () => {
        const str = 'firstKey^firstValue|secondKey^secondValue'

        expect(parseRockKeyValuePairs(str, 'keyOverride', 'valueOverride')).toMatchSnapshot()
    })

    it('parses a key value pair string that contains an empty key value pairing', () => {
        const str = 'firstKey^firstValue|^'

        expect(parseRockKeyValuePairs(str, 'keyOverride', 'valueOverride')).toMatchSnapshot()
    })
  
    it('is passed an empty string for the keyValueStr attribute and returns an empty array', () => {
        expect(parseRockKeyValuePairs('')).toEqual([])
    })

    it('is passed null for the keyValueStr attribute and returns an empty array', () => {
        expect(parseRockKeyValuePairs(null)).toEqual([])
    })
})