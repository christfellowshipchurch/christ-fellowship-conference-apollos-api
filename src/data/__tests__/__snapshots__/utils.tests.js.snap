// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParseRockKeyValuePairs parses a key value pair string that contains an empty key value pairing 1`] = `
Array [
  Object {
    "keyOverride": "firstKey",
    "valueOverride": "firstValue",
  },
  Object {
    "keyOverride": "",
    "valueOverride": "",
  },
]
`;

exports[`ParseRockKeyValuePairs parses a key value pair string with key and value labeling overrides 1`] = `
Array [
  Object {
    "keyOverride": "firstKey",
    "valueOverride": "firstValue",
  },
  Object {
    "keyOverride": "secondKey",
    "valueOverride": "secondValue",
  },
]
`;

exports[`ParseRockKeyValuePairs parses a key value pair string with no overrides 1`] = `
Array [
  Object {
    "key": "firstKey",
    "value": "firstValue",
  },
  Object {
    "key": "secondKey",
    "value": "secondValue",
  },
]
`;
